### DISCUSSION TITLE
Bring back the `<Popper/>` component!

### DISCUSSION THREAD

**Discussion by adri1wald** [mantinedev/mantine#1875] [Feature requests]
Heya,

We have many, many usecases for the `<Popper/>` component in v4. For example, we use it to position menus / toolbars relative to the text selection in our rich text editor.

We made use of the fact that even though the `referenceElement` prop was exposed as type `HTMLDivElement`, the component still works if you pass a `referenceElement` of type [VirtualElement](https://floating-ui.com/docs/virtual-elements) to it.

With v5, it seems this sort of usecase is no longer supported as the exposed components' api require you to pass a react element.

Any chance the `<Popper/>` component could be brought back?

Best,
Adrien

**Comment by rtivital** [comment]
No, I do not think that we will be adding Popper component back – it does not make much sense to export such utility component because none of Mantine components depend on it. You can use an unstyled Popover component instead, or build your own popper with Floating UI hooks.

**Reply by adri1wald** [reply]
Thanks for the response! That makes sense. I guess I'm more asking whether there could be an alternative API for components such as `<Popover/>` and `<Menu/>` that allow you to pass a `referenceElement` that could be a html element or a virtual element, rather than a JSX element

**Reply by rtivital** [reply]
Nope, there are no plans for adding alternative api for these components

**Reply by adri1wald** [reply]
Would you be interested in a contribution towards an alternative API? I think there is a strong case for having such an api when users of mantine want to position a floating UI element relative to the cursor or text selection. Currently I don't see a way to achieve this without reimplementing components such as Menu and Popover in user land.

**Reply by rtivital** [reply]
Currently, I do not think that it is a great idea. I'll possibly add something to handle this use case when I'll be working on the context menu component.
